### ISSUE TITLE
criu check: Does not look good.

### ISSUE THREAD

**Issue by ashish-billore** [checkpoint-restore/criu#1307]
I am trying to run criu inside a container (k8s pod). Earlier, I was able to run it and perform checkpoint and restore inside [dind](https://hub.docker.com/_/docker) so I assume this should also work.

When I run criu check inside the pod, I see following errors/warnings:

```
root@svc-e362ed13-d6ff-47aa-9600-be5057b0a53d-6df955d694-2kqg8:/# criu check
Error (criu/tun.c:85): tun: Unable to create tun: No such file or directory
Warn  (criu/sk-unix.c:229): unix: Unable to open a socket file: Bad address
Error (criu/net.c:3423): Unable create a network namespace: Operation not permitted
Warn  (criu/net.c:3475): NSID isn't reported for network links
Warn  (criu/net.c:3137): Unable to get socket network namespace
Warn  (criu/kerndat.c:869): Can't keep kdat cache on non-tempfs
Error (criu/cr-check.c:146): sys/kernel/ns_last_pid sysctl is inaccessible: Read-only file system
Does not look good.
root@svc-e362ed13-d6ff-47aa-9600-be5057b0a53d-6df955d694-2kqg8:/#
```

I see the host kernel satisfies the prereqs:

```
root@svc-e362ed13-d6ff-47aa-9600-be5057b0a53d-6df955d694-2kqg8:/# uname -a
Linux svc-e362ed13-d6ff-47aa-9600-be5057b0a53d-6df955d694-2kqg8 4.4.247-1.el7.elrepo.x86_64 #1 SMP Tue Dec 1 14:09:51 EST 2020 x86_64 x86_64 x86_64 GNU/Linux
root@svc-e362ed13-d6ff-47aa-9600-be5057b0a53d-6df955d694-2kqg8:/# 
```

So, my query is: what is causing the above errors for criu and how can I fix those?

**Comment by adrianreber** [comment]
This is really hard to answer because there are so many software layers involved running Kubernetes, but in the end it boils done to you not having the necessary privileges. This depends on how it is set up and even you were able to tell us how it is exactly set up I highly doubt we can help you.

> So, my query is: what is causing the above errors for criu and how can I fix those?

In the end, at least for a simple restore, you need to be able to write to `/proc/sys/kernel/ns_last_pid`. If you are able to fix that you are already a step closer to running CRIU. I still expect that your environment has other additional restrictions to make it hard to run CRIU.

**Comment by ashish-billore** [comment]
For now, I was able to get past this issue with privileged containers, so closing this issue.
